### DISCUSSION TITLE
Setting route scoped to dev env, but config in application.rb

### DISCUSSION THREAD

**Discussion by Jackson200** [lookbook-hq/lookbook#145] [General]
Hi there @allmarkedup!

I came across a small gotcha when deploying Lookbook to production.

Following the [Quick Start](https://lookbook.build/guide/quick-start/)  I limited the engine to dev env.

![Screen Shot 2022-09-12 at 9 36 04 AM](https://user-images.githubusercontent.com/39567695/189668529-795e999b-784c-4684-b40a-07eb9782d103.png)

Then I set some [config](https://lookbook.build/guide/config/) in my application.rb

![Screen Shot 2022-09-12 at 9 35 33 AM](https://user-images.githubusercontent.com/39567695/189668770-aaccab2b-490b-47ca-824a-bbb7e0a72e37.png)

Everything worked super in development, but the missing Lookbook::Engine in production, gave me a `lookbook not defined` error from my config/application.rb file.  I easily moved the `config.lookbook` over to  config/development.rb, and it is really obvious.

Still, wondering if it would be better to specify this in the docs. I know there is another section for running in production.
**Thanks again for all your work!**

**Comment by allmarkedup** [comment]
Hey @Jackson200, thanks for this. I have to say that I thought (assumed) that Rails engine configuration was still available regardless of whether the engine itself was mounted or not. Is the Lookbook gem included only in your `development` group or is it available in all environments?

I'll look into this and update the docs regardless as it definitely should be clearer on this topic to prevent people from running into issues when deploying to production.

**Comment by Jackson200** [comment]
Hi @allmarkedup Yes you are right, it's the gem which is only in `development` and when I check back to https://lookbook.build/guide/quick-start/ I see that it was all my doing :) No suggestion of limiting the gem to development.

Maybe I went back and changed it because of the `production` environment warning, and I didn't put 2+2 together!

https://lookbook.build/guide/deployment/
![Screen Shot 2022-09-16 at 10 32 23 AM](https://user-images.githubusercontent.com/39567695/190663827-4762af00-1b3e-4709-aa12-22246460032e.png)

Thanks for all your work and help!

**Comment by allmarkedup** [comment]
@Jackson200 Ok great, thanks for updating me - I will try to add a note clarifying that people need to make sure that the gem is available in production onto the deployment page so others don't get caught out by the same thing 👍
